### DISCUSSION TITLE
PaddleX/latest/pipeline_usage/tutorials/ocr_pipelines/layout_parsing

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleX#2677] [General]
# PaddleX/latest/pipeline_usage/tutorials/ocr_pipelines/layout_parsing

https://paddlepaddle.github.io/PaddleX/latest/pipeline_usage/tutorials/ocr_pipelines/layout_parsing.html

**Comment by 9963KK** [comment]
对于版面分析的模块如果输出我只要layout的输出，不想要ocr的输出，我该怎么做呢？

**Reply by TingquanGao** [reply]
想要layout结果的话可以看下版面解析产线，但目前还不支持禁用ocr功能

**Reply by lhy531** [reply]
哪个版面解析产线，不是就这两个吗？通用版面解析产线和通用解析产线v3，怎么通过服务化的方式只部署版面区域检测单模块？

**Comment by shuaimoumou** [comment]
为什么这个版本不能指定模型？比如我想用V3的模型呢？怎么配置？

**Reply by Ethenlalala** [reply]
配置文件里面不能修改吗

**Reply by shuaimoumou** [reply]
咋改？

**Reply by TingquanGao** [reply]
V3模型正在接入中，如果着急可以先用develop分支，同时设定两个环境变量：PADDLE_PDX_NEW_PREDICTOR=1 USE_NEW_INFERENCE=1 python api_examples/pipelines/test_ocr.py

**Reply by Ethenlalala** [reply]
> 咋改？

通过这个命令获取配置文件，然后在里面改一下试试看
paddlex --get_pipeline_config layout_parsing --save_path ./my_path

**Comment by yijieLiu1** [comment]


**Reply by cuicheng01** [reply]
是一个模型，该模型支持中英文，只是处理英文的话，没有单独的英文模型好，目前PaddleOCR和PaddleX都没有一个模型可以同时支持所有语言的模型，关于支持其他语言的单模型，PaddleX的develop分支已经支持，预计一月底之前拉新的分支

**Comment by we12q** [comment]
我想识别图片里面文字的颜色该怎么做，可以教教我吗。我还是新手

**Reply by cuicheng01** [reply]
可以单独写一段python的脚本来区分不同的颜色，如
```
import cv2
from PIL import Image
import numpy as np

def get_dominant_color(image, k=4):
    # 将图像数据转换为一维数组
    pixels = np.float32(image.reshape(-1, 3))

    # 定义k-means聚类的终止条件
    criteria = (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_MAX_ITER, 100, 0.2)

    # 使用k-means聚类来找到主要颜色
    _, labels, palette = cv2.kmeans(pixels, k, None, criteria, 10, cv2.KMEANS_RANDOM_CENTERS)

    # 计算每个颜色的频率
    _, counts = np.unique(labels, return_counts=True)

    # 找到最常见的颜色
    dominant = palette[np.argmax(counts)]

    return tuple(int(c) for c in dominant)

def detect_text_color(image_path):
    # 使用PIL打开图像并转换为RGB格式
    image = Image.open(image_path).convert('RGB')
    image_np = np.array(image)

    # 假设图像中的文字是最突出（最常见）的颜色之一
    text_color = get_dominant_color(image_np)

    return text_color

if __name__ == "__main__":
    image_path = 'your_image_path_here.png'
    text_color = detect_text_color(image_path)
    print(f"The dominant text color is: {text_color}")

```

**Comment by luckyMe1000** [comment]
请教一下 我想对windows桌面截图进行识别，包括窗体线条和文字，如何选择产线，先用版本解析产线识别窗体线条，再用通用ocr识别文字吗

**Reply by cuicheng01** [reply]
可以考虑直接使用OCR产线，通过位置来筛掉你不感兴趣的区域

**Comment by recapture2018** [comment]
请问通用OCR支持竖直方向字体识别吗？没看见文字方向分类器

**Reply by cuicheng01** [reply]
最新的已经支持了，可以考虑使用文本行方向分类模型，有相关的开关

**Comment by Danee-wawawa** [comment]
您好，想问下SLANet_plus表格识别模型是不是还没有集成到paddleocr的pp-structurev2框架中？

**Reply by ostlee** [reply]
同问, 目前看到v2官方的配置文件里, 加上了有线表和无线表的识别和检测模型, 但是之前的表格结构识别模型好像没有配置说明

**Reply by cuicheng01** [reply]
目前还没有添加，马上发布PP-StructureV3，会附上PP-TableMagic方案，精度更好

**Reply by Danee-wawawa** [reply]
> 目前还没有添加，马上发布PP-StructureV3，会附上PP-TableMagic方案，精度更好

您好，想问下PP-StructureV3是基于paddlex还是paddleocr呢？

**Comment by recapture2018** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by CourageKeenTriumph** [comment]
paddlex.utils.errors.others.UnsupportedParamError: 'RT-DETR-H_layout_17cls' is not a registered model name.
另外paddlepaddle-gpu 和paddlex都是3.0.0rc0版本

**Reply by cuicheng01** [reply]
这个应该是没有安装相关的插件，需要`paddlex --install PaddleDetection`来解决，安装文档中有详细描述哈

**Comment by recapture2018** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by yuanmengwei** [comment]
请问一张pdf要50s吗，有没有快一点的文档识别方法

**Comment by Echo-Forever** [comment]
请问 解析的结果 parsing_res_list 中 没有页码相关信息， 那么block_bbox 的四个坐标如何理解呢  我看下来是递增的 ，是相对于第一页的坐标吗

**Comment by recapture2018** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by recapture2018** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
